src/deploy-builder.ts

```diff
@@ -0,0 +1,42 @@
+import * as path from 'path'
+import * as fs from 'fs-p'
+
+export class DeployBuilder {
```

Comment:
I like that you pulled out this complexity to a separate file but I don't think that the name is intuitive and it doesn't need to be a class.

I'd suggest you export a function called `symlink` instead of a class and use all of the functions outside of the scope of a class.

Does that make sense to you? :)